drivers/espi/espi_mchp_xec_v2.c

```diff
@@ -1561,3 +1561,282 @@ static int espi_xec_init(const struct device *dev)
 
 	return ret;
 }
+
+#ifdef CONFIG_FLASH_HAS_DRIVER_ENABLED
+#undef DT_DRV_COMPAT
+#define DT_DRV_COMPAT microchip_xec_espi_maf_flash
+
+#include <zephyr/drivers/flash.h>
+
+#define MAX_FLASH_REQUEST 64U
+
+static const struct device *const espi_dev = DEVICE_DT_GET(DT_NODELABEL(espi0));
+
+struct flash_espi_data {
+	struct k_sem mutex;
+};
+
+static struct flash_espi_data flash_data;
+
+#ifdef CONFIG_FLASH_PAGE_LAYOUT
+static const struct flash_pages_layout espi_flash_layout[] = {
+	{
+	.pages_size = 4096,
```

Comment:
Magic numbers, add documented macros

Reply:
It would be better if values come from device tree
Also since this driver depends on eSPI node enabled, need to add that relationship in the device tree. 
Will need some input here from dts experts

Reply:
i haven't come up with a solid layout for the EC's flash layout in ESPI yet, so this is still somewhat fluffy and nebulous.  but yes, it should probably come from DT, different board/applications may use different flash layouts.